Node category documentation (#623)
SAP/luigi: docs/navigation-parameters-reference.md

        children: [node, node, node],
        hideFromNav: false,
        isolateView: false,
        icon: 'settings',
        category: {
          label: 'General',
          icon: 'general'

>         }, 

Reviewer: Maybe a comment with an 'or'?
```
category: {
  label: 'General',
  icon: 'general'
},//OR 
category: 'General'
```

Author: Yea, also thought about this, but we didn't have such case until now. Will add one

Author: done

- **loadingIndicator.enabled** shows a loading indicator when switching between micro front-ends. If you have a fast micro front-end, you can disable this feature to prevent flickering of the loading indicator. This parameter is enabled by default.
- **loadingIndicator.hideAutomatically** disables the automatic hiding of the loading indicator once the micro front-end is loaded. It is only considered if the loading indicator is enabled. It does not apply if the loading indicator is activated manually with the `LuigiClient.uxManager().showLoadingIndicator()` function. If the loading indicator is enabled and automatic hiding is disabled, use `LuigiClient.uxManager().hideLoadingIndicator()` to hide it manually in your micro front-end during the startup. This parameter is enabled by default.
- **viewGroup** defines a group of views in the same domain sharing a common security context. This improves performance through reusing the frame. Use **viewGroup** only for the views that use path routing internally.

> - **icon** is the name of an icon, without the `sap-icon--` prefix, from the [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image displayed next to the Node label in the side navigation or instead of the label in the top navigation.

Reviewer 2: ```suggestion
- **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation.

- **hideSideNav** if set to `true`, the left navigation disappears when you click the affected node. It is set to `false` by default.

> - **category** defines a group of views that is separated with a headline and icon. At least one of the nodes of a group should be defined as an Object with `label` and `icon` properties. For all other nodes it is valid to set **category** as string with the `label` value. 

Reviewer 2: ```suggestion
- **category** defines a group of views separated with a headline and an icon. You should define at least one node in a group should as an Object with **label** and **icon** properties. For all other nodes, you can set **category** as a string with the `label` value. 
```

   - **label** is a string that represents the title of the category

>    - **icon** is the name of an icon, without the `sap-icon--` prefix, from the [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image displayed next to the label in the side navigation or instead of the label in the top navigation. In case you accidentially define different icons in a category group, only the first one is used.

Reviewer 2: ```suggestion
   -  **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation. In case you accidentally define different icons in a category group, only the first one is used.
```

- **hideSideNav** if set to `true`, the left navigation disappears when you click the affected node. It is set to `false` by default.
- **category** defines a group of views separated with a headline and an icon. You should define at least one node in a group should as an Object with **label** and **icon** properties. For all other nodes, you can set **category** as a string with the `label` value. 
   - **label** is a string that represents the title of the category
   -  **icon** is the name of an icon, without the `sap-icon--` prefix. Its source may be [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image. The icon is displayed next to the node label in the side navigation or instead of the label in the top navigation. In case you accidentally define different icons in a category group, only the first one is used.

>    - **collapsible** if set to `true`, category items are hidden initially and can be expanded on click. Multiple collapsible categories look like an accordion.

Reviewer 2: ```suggestion
   - **collapsible** if set to `true`, category items are hidden at first. To expand them, click the main category node.
```

- **viewGroup** defines a group of views in the same domain sharing a common security context. This improves performance through reusing the frame. Use **viewGroup** only for the views that use path routing internally.
- **icon** is the name of an icon from the [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image displayed next to the Node label in the side navigation or instead of the label in the top navigation.
- **hideSideNav** if set to `true`, the left navigation disappears when you click the affected node. It is set to `false` by default.
- **category** defines a group of views that is separated with a headline and icon. At least one of the nodes of a group should be defined as an Object with `label` and `icon` properties. For all other nodes it is valid to set **category** as string with the `label` value. 
   - **label** is a string that represents the title of the category

>    - **icon** is the name of an icon from the [OpenUI](https://openui5.hana.ondemand.com/1.40.10/iconExplorer.html) or a custom link (relative or absolute) to an image displayed next to the Node label in the side navigation or instead of the label in the top navigation.

Reviewer 3: Maybe we could explain that for OpenUI icons we already add `sap-icon--` so that the user only needs to specify the remaining part of the icon name.

Reviewer 3: Not sure if it is needed, but we could also say that categories can be used on both left and top navigation to group nodes

Author: good points, added those infos. guess ppl assume automatically it works on top and left nav, so maybe it was more a missing implementation so far
